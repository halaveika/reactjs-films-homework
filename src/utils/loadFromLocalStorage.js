const loadFromLocalStorage = () => {
  const serializedState = localStorage.getItem('state');
  if (serializedState === null) return undefined;
  return JSON.parse(serializedState);
};

export default loadFromLocalStorage;
