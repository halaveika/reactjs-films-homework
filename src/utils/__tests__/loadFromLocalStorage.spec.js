import loadFromLocalStorage from '../loadFromLocalStorage';

const mockState = {
  test: 'TestState'
}

describe('loadFromLocalStorage', () => {

  afterEach(() => {
    jest.restoreAllMocks();
  });
  

  it('should load from localstorage and return object', () => {
    const serializedState = JSON.stringify(mockState);
    localStorage.setItem('teststate', serializedState);
    jest.spyOn(localStorage.__proto__,'getItem').mockImplementation(()=>serializedState)
    const result = loadFromLocalStorage();
    localStorage.removeItem('teststate');
    expect(result).toEqual(mockState);
    expect(result).toMatchSnapshot();
  });

  it('if localstorage is empty should return undefined', () => {
    const result = loadFromLocalStorage();
    expect(result).toBeUndefined();
  });
  
});
