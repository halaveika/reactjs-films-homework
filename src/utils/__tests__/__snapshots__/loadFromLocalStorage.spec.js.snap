// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadFromLocalStorage if localstorage is empty should return undefined 1`] = `undefined`;

exports[`loadFromLocalStorage should load from localstorage and return object 1`] = `
Object {
  "test": "TestState",
}
`;
